dapp/src/DApp.tsx

```diff
@@ -32,24 +33,45 @@ function DApp() {
 }
 
 function DAppProviders() {
+  const [config, setConfig] = useState<Config | undefined>()
+
+  useEffect(() => {
+    const setWagmiConfig = async () => {
+      await delay(500)
```

Comment:
Non blocking:
Such time based approach seems prone to cause unexpected issues. 
What if 500ms is too little? Have you tried [asynchronous dynamic imports](https://vite.dev/guide/features#dynamic-import) for this issue? 

Looks like it might work here better.

Reply:
The delay is only to display the "loading" screen for a bit longer than a quick flash so that it is noticeable for a moment. See the `getWagmiConfig` function. It uses dynamic import.